### PR TITLE
feat: [CU-860rmfpv8] drag and drop ships

### PR DESCRIPTION
**Pull request by MASBERNATQ**
## Description

Drag and drop the ships in the board.

## Changelog

- Install `react-dnd` and `react-dnd-touch-backend` dependencies
- Drag and drop ships in the board
- Temporarily hide start zone for ships (will be done in another ticket after better reflection of the thing)
- Rename `get` radash function to `getItem` 

## Related ticket

This pull request is related to ClickUp ticket : **[CU-860rmfpv8](https://app.clickup.com/t/860rmfpv8)**.

## Checklist

- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have named the title of my pull request while respecting the naming convention:

```
<type>(<scope>): [<ticket id>] <short summary>
  │       │            │              │
  │       │            │              └─ Summary in present tense. Not capitalized. No period at the end.
  │       │            │
  │       │            └─ ClickUp ticket id (optional)
  │       │
  │       └─ Commit Scope (optional)
  │
  └─ Commit Type: build|chore|ci|docs|feat|fix|perf|refactor|revert|style|test
```

### CODE REVIEW

**Comment by FlavianEng** [comment]
Task linked: [CU-860rmfpv8 US: As a player, on ships placement view, I'd like to drag and drop my ships](https://app.clickup.com/t/860rmfpv8)

File: src/ship/components/Board/Board.tsx [hunk 1]
```diff
@@ -1,13 +1,17 @@
 import type { CSSProperties } from 'react';
+import { useRef } from 'react';
 import { useCallback, useMemo } from 'react';
+import { useDrop } from 'react-dnd';
 import classNames from 'clsx';
+import { get } from 'radash';
```
Comment (FlavianEng) [inline src/ship/components/Board/Board.tsx:6]: `get` is ambiguous, can you rename the dependency ?
```suggestion
import { get as radashGet } from 'radash';
```

Comment (MASBERNATQ) [reply src/ship/components/Board/Board.tsx]: :heavy_check_mark:

File: src/ship/components/Board/Board.tsx [hunk 2]
```diff
@@ -29,6 +34,38 @@ function Board(props: BoardProps): JSX.Element {
...
+  const boardRef = useRef<HTMLTableElement>(null);
+  const cellSize = boardRef.current
+    ? boardRef.current.getBoundingClientRect().width / dimensions
+    : 0;
+
+  const [, drop] = useDrop(
+    () => ({
+      accept: SHIP_TYPE,
+      canDrop: () => isPlacementActive,
+      drop: (item: BoardShipPlacementDropped, monitor): void => {
```
[the remark this answers is not in the document]
Comment (MASBERNATQ) [reply src/ship/components/Board/Board.tsx:46]: C'est ce qu'il utilise pour gérer tout le truc en interne.

**Review by FlavianEng — APPROVED** [review]

**Pull request merged**